5.drug-dependency/scripts/4.drug_latent_dim_correlation.py

```
@@ -0,0 +1,263 @@
+        if not combined_results_df.empty:
+            if ((combined_results_df['model'] == model_name) & 
+                (combined_results_df['full_model_z'] == num_components)).any():
+                print(f"Skipping {model_name} with {num_components} dimensions as it is already processed.")
+                continue  # Skip to the next iteration if this combination is already present
+        
+        # Load the saved model
+        model_filename = model_save_dir / f"{model_name}_{num_components}_components_model.joblib"
+        if model_filename.exists():
+            print(f"Loading model from {model_filename}")
+            model = joblib.load(model_filename)
+            
+            if model_name in ["pca", "ica", "nmf"]:
+                # Extract the latent dimensions for these models
+                latent_df = sklearn_extract_latent_dimensions(model, model_name, dependency_df)
+            elif model_name == "betavae":
+                latent_df = extract_latent_dimensions(model, train_and_test_subbed_loader, metadata)
+            elif model_name == "betatcvae":
+                latent_df = tc_extract_latent_dimensions(model, train_and_test_subbed_loader, metadata)
+            elif model_name == "vanillavae":
+                latent_df = vvae_extract_latent_dimensions(model, train_and_test_subbed_loader, metadata)
+
+            latent_df.columns = ['ModelID'] + [f'z_{col}' if isinstance(col, int) else col for col in latent_df.columns[1:]]
+            # Perform Pearson correlation between latent dimensions and drug data
+            correlation_results_df = perform_correlation(latent_df, prism_df, model_name, num_components)
+            
+            # Concatenate results to the combined dataframe
+            combined_results_df = pd.concat([combined_results_df, correlation_results_df], ignore_index=True)
+        else:
+            print(f"Model file {model_filename} not found. Skipping.")
```

Should the script fail if the model is not present? Dont we want all models ran?

---

Yes, fixed